shell/platform/android/ndk_helpers.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/android/ndk_helpers.h"
+
+#include "shell/platform/android/ndk_helpers.h"
+
+#include "flutter/fml/logging.h"
+
+#include <dlfcn.h>
+
+namespace flutter {
+
+typedef AHardwareBuffer* (*fp_AHardwareBuffer_fromHardwareBuffer)(
+    JNIEnv* env,
+    jobject hardwareBufferObj);
+typedef void (*fp_AHardwareBuffer_acquire)(AHardwareBuffer* buffer);
+typedef void (*fp_AHardwareBuffer_release)(AHardwareBuffer* buffer);
+typedef EGLClientBuffer (*fp_eglGetNativeClientBufferANDROID)(
+    AHardwareBuffer* buffer);
+
+static AHardwareBuffer* (*_AHardwareBuffer_fromHardwareBuffer)(
+    JNIEnv* env,
+    jobject hardwareBufferObj) = nullptr;
+static void (*_AHardwareBuffer_acquire)(AHardwareBuffer* buffer) = nullptr;
+static void (*_AHardwareBuffer_release)(AHardwareBuffer* buffer) = nullptr;
+static EGLClientBuffer (*_eglGetNativeClientBufferANDROID)(
+    AHardwareBuffer* buffer) = nullptr;
+
+void NDKHelpers::Init() {
+  FML_LOG(INFO) << "NDKHelper::Init";
+  {
+    void* lib = dlopen("libEGL.so", RTLD_NOW | RTLD_LOCAL);
```

Comment:
Consider using `fml::NativeLibrary` for this. It may have some fixes for common issues (but is likely just what you have here TBH).